### DISCUSSION TITLE
Text to Widget Importer

### DISCUSSION THREAD

**Discussion by SeitenDesign** [elementor/elementor#24558] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

A faster workflow when building websites.

### Describe the solution you'd like

A significant part of my job involves populating my clients' websites with content. Accordions, in particular, are a major time sink. It would be great if accordions, and possibly other widgets, had an import function. Suppose I have a Word document with headings and paragraphs; it would be cool if I could import this into the accordion widget, and the widget would then generate an accordion from it. All that would be left to do is the styling, and within a few seconds, you could have an accordion with 20 entries.

**Comment by hitsgee** [comment]
I would like to see this feature as well. For now I use a clipboard program that allows you to copy up to 30 items such as titles or paragraphs and then paste sequentially in order. Much faster than manually copying and pasting every time.

**Comment by m1ga** [comment]
Elementor has a JS API: https://developers.elementor.com/js-api/js-api-commands/
E.g. this command will create a container when you run it in the editor:
```js
$e.run('document/elements/create',
	{
		model: { elType: 'container' },
		columns: 1,
		container: elementor.getContainer( 'document' ),
	},
);
```

so you could try and run a puppeteer script that reads your word (https://www.npmjs.com/package/textract) file, opens your page and uses the JS API to create elements for you. Not tried it but you could look at the Elementor tests where they use some API calls for testing https://github.com/elementor/elementor/tree/main/tests/playwright to see if you can reuse that to write yourself a "bot"
